inline/src/lexer/mod.rs

```diff
@@ -549,14 +436,14 @@ impl TokenIterator<'_> {
         }
 
         // NOTE: index points to the NEXT character, token Span is UP TO that character
-        let offset = self.index - start_pos.column;
-        let end_pos = self.pos + (0, offset);
+        let end_pos = self.get_symbol(self.index - 1)?.end;
 
-        let len = self.index - start_pos.column.saturating_sub(1);
+        let span = Span::from((start_pos, end_pos));
+        let len = span.len_grapheme()?;
 
         // TODO: improve this logic
```

Comment:
as internally discussed, this TODO should not be relevant anymore => TODO and commented code can be removed